pkg/secrets_reflector/secrets_reflector_test.go

```diff
@@ -0,0 +1,246 @@
+package secretsreflector
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/dcos/client-go/dcos"
+	logtest "github.com/sirupsen/logrus/hooks/test"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	kubeerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+
+	dklbcache "github.com/mesosphere/dklb/pkg/cache"
+	"github.com/mesosphere/dklb/pkg/constants"
+	cachetestutil "github.com/mesosphere/dklb/test/util/cache"
+	secrettestutil "github.com/mesosphere/dklb/test/util/kubernetes/secret"
+)
+
+var (
+	defaultTestKubeSecret = secrettestutil.DummySecretResource("namespace-1", "name-1", func(secret *corev1.Secret) {
+		secret.Data["tls.crt"] = []byte("aGVsbG8gd29ybGQK")
+		secret.Data["tls.key"] = []byte("aGVsbG8gd29ybGQK")
+	})
+	defaultTestLogger, defaultTestLoggerHook = logtest.NewNullLogger()
+)
+
+func TestSecretReflector_translate(t *testing.T) {
+	tests := []struct {
+		description    string
+		expectedError  error
+		expectedResult *dcos.SecretsV1Secret
+		secret         *corev1.Secret
+	}{
+		{
+			description:   "should translate a secret",
+			expectedError: nil,
+			expectedResult: &dcos.SecretsV1Secret{
+				Value: "hello world\nhello world\n",
```

Comment:
As above, let's use distinct values for cert and key so we are sure correct fields are used in correct order. So this would read more like `hello world\ngoodbye world\n` for example.